compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -78,12 +105,48 @@ fn defaultness(tcx: TyCtxt<'_>, def_id: LocalDefId) -> hir::Defaultness {
 /// Calculates the `Sized` constraint.
 ///
 /// In fact, there are only a few options for the types in the constraint:
-///     - an obviously-unsized type
+///     - an metasized type
+///     - an pointee-sized type
 ///     - a type parameter or projection whose sizedness can't be known
 #[instrument(level = "debug", skip(tcx), ret)]
 fn adt_sized_constraint<'tcx>(
     tcx: TyCtxt<'tcx>,
     def_id: DefId,
+) -> Option<ty::EarlyBinder<'tcx, Ty<'tcx>>> {
+    adt_sizedness_constraint(tcx, Sizedness::Sized, def_id)
+}
+
+/// Calculates the `MetaSized` constraint.
+///
+/// In fact, there are only a few options for the types in the constraint:
+///     - an pointeesized type
+///     - a type parameter or projection whose sizedness can't be known
+#[instrument(level = "debug", skip(tcx), ret)]
+fn adt_metasized_constraint<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    def_id: DefId,
+) -> Option<ty::EarlyBinder<'tcx, Ty<'tcx>>> {
+    adt_sizedness_constraint(tcx, Sizedness::MetaSized, def_id)
+}
+
+///
+/// Calculates the `PointeeSized` constraint.
+///
+/// In fact, there are only a few options for the types in the constraint:
+///     - a type parameter or projection whose sizedness can't be known
```

Comment:
this should be reworded now that there's only one case

Reply:
Fixed